worksheet.rb

```diff
@@ -31,4 +42,134 @@
 # - the total amount of money each driver has made
 # - the average rating for each driver
 # - Which driver made the most money?
-# - Which driver has the highest average rating?
\ No newline at end of file
+# - Which driver has the highest average rating?
+
+driver_records = {
+    DR0001: [
+        {
+            date: "3rd Feb 2016",
+            cost: 10,
+            rider: "RD0003",
+            rating: 3
+        },
+        {
+            date: "3rd Feb 2016",
+            cost: 30,
+            rider: "RD0015",
+            rating: 4
+        },
+        {
+            date: "5th Feb 2016",
+            cost: 45,
+            rider: "RD0003",
+            rating: 2
+        }
+    ],
+    DR0002: [
+        {
+            date: "3rd Feb 2016",
+            cost: 25,
+            rider: "RD0073",
+            rating: 5
+        },
+        {
+            date: "4th Feb 2016",
+            cost: 15,
+            rider: "RD0013",
+            rating: 1
+        },
+        {
+            date: "5th Feb 2016",
+            cost: 35,
+            rider: "RD0066",
+            rating: 3
+        }
+    ],
+    DR0003: [
+        {
+            date: "4th Feb 2016",
+            cost: 5,
+            rider: "RD0066",
+            rating: 5
+        },
+        {
+            date: "5th Feb 2016",
+            cost: 50,
+            rider: "RD0003",
+            rating: 2
+        }
+    ],
+    DR0004: [
+        {
+            date: "3rd Feb 2016",
+            cost: 5,
+            rider: "RD0022",
+            rating: 5
+        },
+        {
+            date: "4th Feb 2016",
+            cost: 10,
+            rider: "RD0022",
+            rating: 4
+        },
+        {
+            date: "5th Feb 2016",
+            cost: 20,
+            rider: "RD0073",
+            rating: 5
+        }
+    ]
+}
+
+# - the number of rides each driver has given
+def sum_rides(rides_list)
+  rides_list.each do |driver, trip|
+    puts "#{driver} gave #{trip.length} rides"
+  end
+  return rides_list
+end
+
+# - the total amount of money each driver has made
+def sum_money(total_cost)
+  total_earned = 0
+
+  total_cost.each do |driver, cost_value|
+    cost_value.each do |earning|
+      trip_cost = earning[:cost]
+      total_earned += trip_cost
+    end
+    puts "#{driver} made $#{'%.2f' % total_earned}"
+    total_earned = 0
+  end
+  return total_earned
+end
+
+# - the average rating for each driver
+def find_average(driver_rating)
+  rating_total = 0
+  rating_average = 0
+
+  driver_rating.each do |driver, rating|
+    rating.each do |review|
+      rider_review = review[:rating]
+      rating_total += rider_review
+      rating_average = rating_total / rating.length
+    end
+    puts "#{driver} average rating: #{rating_average}"
+    rating_total = 0
+  end
+  return rating_average
+end
+
+# - Which driver made the most money?
+# - Which driver has the highest average rating?
```

Comment:
You have done all the work above to make answering these questions possible. If you call the methods above for all the drivers, you could collect the data in an array, and then find the max. Let me know if you'd like to go over this!